Extensions/Today/TodayViewController.swift

```diff
@@ -171,26 +171,13 @@ class TodayViewController: UIViewController, NCWidgetProviding {
         } else {
             self.openCopiedLinkButton.hidden = true
         }
-
-        self.view.setNeedsLayout()
     }
 
     func widgetMarginInsetsForProposedMarginInsets(defaultMarginInsets: UIEdgeInsets) -> UIEdgeInsets {
         TodayUX.defaultWidgetTextMargin = defaultMarginInsets.left
         return UIEdgeInsetsMake(0, 0, TodayUX.verticalWidgetMargin, 0)
     }
 
-    func widgetPerformUpdateWithCompletionHandler(completionHandler: ((NCUpdateResult) -> Void)) {
-        // Perform any setup necessary in order to update the view.
-        dispatch_async(dispatch_get_main_queue()) {
-            // updates need to be made on the main thread
-            self.updateCopiedLink()
-            // and we need to call the completion handler in every branch.
-            completionHandler(NCUpdateResult.NewData)
-        }
-        completionHandler(NCUpdateResult.NewData)
-    }
-
```

Comment:
This looks good now. 

However I'm seeing the link justify to the right when the link is longer than the button width.